grammar.ebnf

```diff
@@ -33,14 +42,54 @@ Statement = "{", {Statement}, "}"
   because we can achieve that as:
   Arr arr = a.b;  // arr is a pointer under the hood - any change to its
   elements will be reflected to a.b arr = a.b;
-*/
+*)
+
 LValue = Identifier, [ ".", Identifier ];
 
+Expression = FunctionCall
+           | NotExpression
+           | ArrayLookupExpression
+           | BinaryExpression;
```

Comment:
Those should have a precedence as well. For example, the `[]` should be of higher precedence than `+`.